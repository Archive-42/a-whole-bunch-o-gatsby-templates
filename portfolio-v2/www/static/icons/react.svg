<svg width="50" height="50" fill="none" xmlns="http://www.w3.org/2000/svg"><rect width="50" height="50" rx="6" fill="#ECFBFF"/><path fill-rule="evenodd" clip-rule="evenodd" d="M36.014 18.555c4.833 1.48 7.986 3.87 7.986 6.39 0 2.513-3.137 4.901-7.94 6.382 1.124 4.917.636 8.833-1.54 10.09-.512.294-1.1.434-1.751.434-2.13 0-4.919-1.505-7.738-4.126-2.82 2.637-5.608 4.157-7.738 4.157-.65 0-1.24-.14-1.743-.434-2.184-1.257-2.68-5.18-1.564-10.113C9.152 29.854 6 27.465 6 24.945c0-2.513 3.137-4.894 7.955-6.383-1.123-4.916-.635-8.833 1.541-10.089.511-.295 1.1-.434 1.75-.434V8.03c2.13 0 4.92 1.505 7.739 4.126C27.803 9.52 30.591 8 32.721 8v.008c.64 0 1.226.14 1.728.434 2.184 1.256 2.68 5.18 1.564 10.113zm-3.3-8.826c-1.61.005-3.977 1.292-6.498 3.638a37.815 37.815 0 013.222 3.877c1.743.186 3.4.473 4.919.845.17-.783.31-1.55.395-2.28.356-2.962-.101-5.265-1.154-5.87-.24-.132-.527-.202-.883-.202V9.73zm-8.961 3.645C21.228 11.04 18.85 9.76 17.247 9.76v.008c-.357 0-.659.062-.891.202-1.046.612-1.495 2.916-1.131 5.878.093.721.225 1.474.403 2.25a36.126 36.126 0 014.926-.854 36.62 36.62 0 013.199-3.87zm9.117 9.461a34.94 34.94 0 001.045-3.063 34.915 34.915 0 00-3.175-.612c.379.589.759 1.194 1.115 1.814.364.62.697 1.241 1.015 1.861zM28.5 30.995a41.294 41.294 0 003.47-6.057 40.458 40.458 0 00-1.611-3.086 42.749 42.749 0 00-1.875-2.947 40.26 40.26 0 00-6.986.008 41.294 41.294 0 00-3.47 6.056 40.458 40.458 0 001.611 3.087 42.749 42.749 0 001.875 2.947 40.26 40.26 0 006.986-.008zm5.437-.884a33.23 33.23 0 00-1.069-3.087c-.317.62-.65 1.249-1.006 1.87-.357.627-.736 1.24-1.116 1.837a34.717 34.717 0 003.191-.62zm-11.076 2.706a31.964 31.964 0 002.154 2.482 30.252 30.252 0 002.13-2.482c-.705.04-1.426.055-2.154.055-.72 0-1.433-.024-2.13-.055zm-6.785-2.698c1.015.248 2.076.45 3.176.612-.38-.59-.752-1.194-1.116-1.814-.364-.62-.697-1.241-1.014-1.862a34.94 34.94 0 00-1.046 3.064zM27.13 17.074a31.963 31.963 0 00-2.153-2.482 30.253 30.253 0 00-2.13 2.481A39.285 39.285 0 0125 17.02c.72 0 1.433.023 2.13.055zm-9 3.924c.356-.629.736-1.241 1.115-1.838a37.56 37.56 0 00-3.191.612c.294 1.008.65 2.048 1.069 3.087.317-.62.65-1.241 1.007-1.861zM7.72 24.945c0 1.217 1.773 2.753 4.515 3.924.674.287 1.402.55 2.169.791a36.761 36.761 0 011.72-4.7 37.28 37.28 0 01-1.743-4.722c-.752.24-1.48.496-2.146.783-2.742 1.163-4.515 2.706-4.515 3.924zm7.528 9.135c-.356 2.963.101 5.266 1.154 5.871.233.132.527.202.876.21 1.61 0 3.98-1.288 6.506-3.638a37.815 37.815 0 01-3.222-3.877 37.48 37.48 0 01-4.919-.846c-.17.784-.31 1.551-.395 2.28zm18.396 5.84c1.046-.613 1.495-2.916 1.131-5.878a21.75 21.75 0 00-.403-2.257 36.14 36.14 0 01-4.926.853 36.62 36.62 0 01-3.199 3.87c2.525 2.334 4.903 3.614 6.506 3.614.357 0 .659-.062.891-.202zm1.968-10.268c.75-.24 1.479-.496 2.145-.783 2.742-1.163 4.516-2.707 4.524-3.924 0-1.218-1.774-2.753-4.516-3.924-.682-.287-1.41-.55-2.177-.791a36.762 36.762 0 01-1.72 4.7 37.266 37.266 0 011.743 4.722zm-7.08-4.707a3.542 3.542 0 01-3.54 3.544 3.542 3.542 0 01-3.54-3.544 3.542 3.542 0 013.54-3.544 3.542 3.542 0 013.54 3.544z" fill="#61DAFB"/></svg>
