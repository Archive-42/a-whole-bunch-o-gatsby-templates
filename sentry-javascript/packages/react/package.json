{
  "name": "@sentry/react",
  "version": "6.2.5",
  "description": "Offical Sentry SDK for React.js",
  "repository": "git://github.com/getsentry/sentry-javascript.git",
  "homepage": "https://github.com/getsentry/sentry-javascript/tree/master/packages/react",
  "author": "Sentry",
  "license": "BSD-3-Clause",
  "engines": {
    "node": ">=6"
  },
  "main": "dist/index.js",
  "module": "esm/index.js",
  "types": "dist/index.d.ts",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@sentry/browser": "6.2.5",
    "@sentry/minimal": "6.2.5",
    "@sentry/types": "6.2.5",
    "@sentry/utils": "6.2.5",
    "hoist-non-react-statics": "^3.3.2",
    "tslib": "^1.9.3"
  },
  "peerDependencies": {
    "react": "15.x || 16.x || 17.x"
  },
  "devDependencies": {
    "@sentry-internal/eslint-config-sdk": "6.2.5",
    "@testing-library/react": "^11.2.6",
    "@testing-library/react-hooks": "^5.1.1",
    "@types/history-4": "npm:@types/history@4.7.8",
    "@types/history-5": "npm:@types/history@4.7.8",
    "@types/hoist-non-react-statics": "^3.3.1",
    "@types/react": "^17.0.3",
    "@types/react-router-3": "npm:@types/react-router@3.0.24",
    "@types/react-router-4": "npm:@types/react-router@5.1.13",
    "@types/react-router-5": "npm:@types/react-router@5.1.13",
    "eslint": "7.6.0",
    "eslint-plugin-react": "^7.20.5",
    "eslint-plugin-react-hooks": "^4.0.8",
    "history-4": "npm:history@4.6.0",
    "history-5": "npm:history@4.9.0",
    "jest": "^24.7.1",
    "jsdom": "^16.2.2",
    "npm-run-all": "^4.1.2",
    "prettier": "1.19.0",
    "react": "^16.0.0",
    "react-dom": "^16.0.0",
    "react-router-3": "npm:react-router@3.2.0",
    "react-router-4": "npm:react-router@4.1.0",
    "react-router-5": "npm:react-router@5.0.0",
    "react-test-renderer": "^16.13.1",
    "redux": "^4.0.5",
    "rimraf": "^2.6.3",
    "typescript": "3.7.5"
  },
  "scripts": {
    "build": "run-p build:es5 build:esm",
    "build:es5": "tsc -p tsconfig.build.json",
    "build:esm": "tsc -p tsconfig.esm.json",
    "build:watch": "run-p build:watch:es5 build:watch:esm",
    "build:watch:es5": "tsc -p tsconfig.build.json -w --preserveWatchOutput",
    "build:watch:esm": "tsc -p tsconfig.esm.json -w --preserveWatchOutput",
    "clean": "rimraf dist coverage build esm",
    "link:yarn": "yarn link",
    "lint": "run-s lint:prettier lint:eslint",
    "lint:prettier": "prettier --check \"{src,test}/**/*.ts\"",
    "lint:eslint": "eslint . --cache --cache-location '../../eslintcache/' --format stylish",
    "fix": "run-s fix:eslint fix:prettier",
    "fix:prettier": "prettier --write \"{src,test}/**/*.ts\"",
    "fix:eslint": "eslint . --format stylish --fix",
    "test": "jest",
    "test:watch": "jest --watch",
    "pack": "npm pack"
  },
  "volta": {
    "extends": "../../package.json"
  },
  "jest": {
    "collectCoverage": true,
    "transform": {
      "^.+\\.ts$": "ts-jest",
      "^.+\\.tsx$": "ts-jest"
    },
    "moduleFileExtensions": [
      "js",
      "ts",
      "tsx"
    ],
    "testEnvironment": "jsdom",
    "testMatch": [
      "**/*.test.ts",
      "**/*.test.tsx"
    ],
    "globals": {
      "ts-jest": {
        "tsConfig": "./tsconfig.json",
        "diagnostics": false
      }
    }
  },
  "sideEffects": [
    "./src/index.ts"
  ]
}
