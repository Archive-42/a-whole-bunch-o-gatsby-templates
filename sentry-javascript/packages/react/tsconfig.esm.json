{
  "extends": "../../tsconfig.esm.json",
  "compilerOptions": {
    "esModuleInterop": true,
    "baseUrl": ".",
    "outDir": "esm",
    "jsx": "react"
  },
  "include": ["src/**/*"]
}
