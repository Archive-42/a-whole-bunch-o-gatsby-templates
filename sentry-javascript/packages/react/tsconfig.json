{
  "extends": "./tsconfig.build.json",
  "include": ["src/**/*.ts", "test/**/*.ts", "src/**/*.tsx", "test/**/*.tsx"],
  "exclude": ["dist"],
  "compilerOptions": {
    "rootDir": ".",
    "types": ["jest"]
  }
}
