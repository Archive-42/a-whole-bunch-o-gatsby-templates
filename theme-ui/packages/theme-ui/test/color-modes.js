/** @jsx jsx */
import React from 'react'
import renderer from 'react-test-renderer'
import { render, fireEvent, cleanup, act } from '@testing-library/react'
import { matchers } from 'jest-emotion'
import {
  jsx,
  ThemeProvider,
  useColorMode,
  useThemeUI,
  ColorMode,
} from '../src/index'

const STORAGE_KEY = 'theme-ui-color-mode'

afterEach(cleanup)
beforeEach(() => {
  localStorage.removeItem(STORAGE_KEY)
})
expect.extend(matchers)

test('renders with color modes', () => {
  let json
  let mode
  const Mode = props => {
    const [colorMode] = useColorMode()
    mode = colorMode
    return <div>Mode</div>
  }
  renderer.act(() => {
    renderer.create(
      <ThemeProvider
        theme={{
          initialColorMode: 'light',
        }}>
        <Mode />
      </ThemeProvider>
    )
  })
  expect(mode).toBe('light')
})

test('useColorMode updates color mode state', () => {
  let mode
  const Button = props => {
    const [colorMode, setMode] = useColorMode()
    mode = colorMode
    return (
      <button
        onClick={e => {
          setMode('dark')
        }}
        children="test"
      />
    )
  }
  const tree = render(
    <ThemeProvider>
      <Button />
    </ThemeProvider>
  )
  const button = tree.getByText('test')
  fireEvent.click(button)
  expect(mode).toBe('dark')
})

test('color mode is passed through theme context', () => {
  let mode
  const Button = props => {
    const [colorMode, setMode] = useColorMode()
    mode = colorMode
    return (
      <button
        sx={{
          color: 'text',
        }}
        onClick={e => {
          setMode('dark')
        }}
        children="test"
      />
    )
  }
  const tree = render(
    <ThemeProvider
      theme={{
        colors: {
          text: '#000',
          modes: {
            dark: {
              text: 'cyan',
            },
          },
        },
      }}>
      <Button />
    </ThemeProvider>
  )
  const button = tree.getByText('test')
  button.click()
  expect(mode).toBe('dark')
  expect(tree.getByText('test')).toHaveStyleRule('color', 'cyan')
})

test('converts color modes to css properties', () => {
  const Box = props => (
    <div
      sx={{
        color: 'text',
      }}
      children="test"
    />
  )
  const tree = render(
    <ThemeProvider
      theme={{
        useCustomProperties: true,
        colors: {
          text: '#000',
          modes: {
            dark: {
              text: '#fff',
            },
          },
        },
      }}>
      <Box />
    </ThemeProvider>
  )
  expect(tree.getByText('test')).toHaveStyleRule(
    'color',
    'var(--theme-ui-text,#000)'
  )
})

test('does not initialize mode', () => {
  let mode
  const Button = props => {
    const [colorMode, setMode] = useColorMode()
    mode = colorMode
    return <button children="test" />
  }
  const tree = render(
    <ThemeProvider>
      <Button />
    </ThemeProvider>
  )
  expect(mode).toBe(undefined)
})

test('initializes mode based on localStorage', () => {
  window.localStorage.setItem(STORAGE_KEY, 'dark')
  let mode
  const Button = props => {
    const [colorMode, setMode] = useColorMode()
    mode = colorMode
    return <button children="test" />
  }
  const tree = render(
    <ThemeProvider>
      <Button />
    </ThemeProvider>
  )
  expect(mode).toBe('dark')
})

test('inherits color mode state from parent context', () => {
  let mode
  const Consumer = props => {
    const [colorMode] = useColorMode()
    mode = colorMode
    return false
  }
  render(
    <ThemeProvider
      theme={{
        initialColorMode: 'outer',
      }}>
      <ThemeProvider
        theme={{
          initialColorMode: 'inner',
        }}>
        <Consumer />
      </ThemeProvider>
    </ThemeProvider>
  )
  expect(mode).toBe('outer')
})

test('retains initial context', () => {
  let context
  const Consumer = props => {
    context = useThemeUI()
    return false
  }
  render(
    <ThemeProvider>
      <Consumer />
    </ThemeProvider>
  )
  expect(typeof context.components).toBe('object')
  expect(context.components.h1).toBeTruthy()
  expect(context.components.pre).toBeTruthy()
  expect(context.components.blockquote).toBeTruthy()
})

test('initializes mode from prefers-color-scheme media query', () => {
  window.matchMedia = jest.fn().mockImplementation(query => {
    return {
      matches: true,
      media: query,
    }
  })
  let mode
  const Consumer = props => {
    const [colorMode] = useColorMode()
    mode = colorMode
    return false
  }
  render(
    <ThemeProvider>
      <Consumer />
    </ThemeProvider>
  )
  expect(mode).toBe('dark')
})

test('does not initialize mode from prefers-color-scheme media query', () => {
  window.matchMedia = jest.fn().mockImplementation(query => {
    return {
      matches: false,
      media: query,
    }
  })
  let mode
  const Consumer = props => {
    const [colorMode] = useColorMode()
    mode = colorMode
    return false
  }
  render(
    <ThemeProvider>
      <Consumer />
    </ThemeProvider>
  )
  expect(mode).toBe(undefined)
})

test('ColorMode component renders null', () => {
  const json = renderer
    .create(
      <ThemeProvider>
        <ColorMode />
      </ThemeProvider>
    )
    .toJSON()
  expect(json).toBe(null)
})

test('ColorMode component renders with colors', () => {
  const root = render(
    <ThemeProvider
      theme={{
        colors: {
          text: 'tomato',
          background: 'black',
          modes: {
            tomato: {
              text: 'black',
              background: 'tomato',
            },
          },
        },
      }}>
      <ColorMode />
    </ThemeProvider>
  )
  const styles = document.querySelector('style').innerHTML
  expect(styles).toMatchSnapshot()
})

test('useColorMode throws when there is no theme context', () => {
  expect(() => {
    const Consumer = props => {
      const [colorMode] = useColorMode('beep')
      mode = colorMode
      return false
    }
    render(<Consumer />)
  }).toThrow()
})

test('useThemeUI returns current color mode colors', () => {
  window.localStorage.setItem(STORAGE_KEY, 'tomato')
  let colors
  const GetColors = () => {
    const { theme } = useThemeUI()
    colors = theme.colors
    return false
  }
  const root = render(
    <ThemeProvider
      theme={{
        initialColorMode: 'light',
        colors: {
          text: 'tomato',
          background: 'black',
          modes: {
            tomato: {
              text: 'black',
              background: 'tomato',
            },
          },
        },
      }}>
      <GetColors />
    </ThemeProvider>
  )
  expect(colors.text).toBe('black')
  expect(colors.background).toBe('tomato')
})
