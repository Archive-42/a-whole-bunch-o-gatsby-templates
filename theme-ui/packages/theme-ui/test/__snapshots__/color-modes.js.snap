// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorMode component renders with colors 1`] = `"body{color:tomato;background-color:black;}"`;
